internal/workload/manager.go

```diff
@@ -475,6 +477,26 @@ func (w *WorkloadManager) toPod(workload *models.Workload) (*v1.Pod, error) {
 		containers = append(containers, container)
 	}
 	pod.Spec.Containers = containers
+
+	if pod.Labels == nil {
+		pod.Labels = map[string]string{}
+	}
+
+	// Set the authfile label to pod, if ImageRegistry authfile is set:
+	if workload.ImageRegistries != nil && workload.ImageRegistries.AuthFile != "" {
+		pod.Labels["io.containers.autoupdate.authfile"] = workload.ImageRegistries.AuthFile
```

Comment:
Is `io.containers.autoupdate.authfile` supposed to be a path to auth file or should it be the authfile JSON? `workload.ImageRegistries.AuthFile` is the JSON. File path can be calculated with https://github.com/project-flotta/flotta-device-worker/blob/4bf1eb43a94fdc32d86d88580417e2f946e0fa81/internal/workload/manager.go#L257.

Reply:
Nice catch!